### ISSUE TITLE
Create Boundary Tilesets

### ISSUE THREAD

**Issue by cKellyDesign** [onaio/reveal-frontend#8]
Gisida will use tilesets hosted on Mapbox Studio to render Administrative boundaries as line layers.

**Comment by cKellyDesign** [comment]
We are using different combinations of Tilsets and Geojson for both Admin and non-Admin Jurisdictions in different contexts and which we need to clean up and define implementation documentation.

This issue has potential to provide substantial performance benefits if we can leverage tilesets as much as possible.

**Comment by mberg** [comment]
@cKellyDesign we should use our own layers for admin 1 and 2 from a site like humdata or it will be provided by the client.  

Can we use tilesets for all layers to speed things up.  Right now the load times due to loading village layers is very slow.  Can we discuss if there is not a reason just to use tilesets now.  We can easily create these in mapbox.  Help me understand what I'm missing please.

**Comment by moshthepitt** [comment]
@mberg the thing that is not clear to me about using tilesets is:

- How do we configure these for the web app in a sane way?  Using geojson from OpenSRP is nice and easy because we just tell the web app where the API endpoint is and it can load everything nicely.  It is not clear to me how we would handle this for tilesets.  Will we have a giant config file that has every single layer mapped to an OpenSRP jurisdiction id?  How easy will that be to manage?

**Comment by cKellyDesign** [comment]
> we should use our own layers for admin 1 and 2 from a site like humdata
@mberg 
Hard and requirements for this:
R1. the data of a tileset must match at least one attribute uniquely identifying the location and at least one attribute uniquely identifying the parent location
R2. the attributes values of `R1` must match that which is in OpenSRP (and OpenMRS)
R3. ideally the attributes described in `R1` match what is in OpenSRP, otherwise we need to specify configurations telling the app how to identify tileset features as locations 

Namibia Villages:
N1. utilizing a tileset for this layer will negate the need to download location geojson
N2. the tileset we have has data values which do not match that which is in OpenSRP (`VILLAGE NAME` in tileset vs `Village Name` in OpenSRP violates `R2`)
N3. updating the shape file and tileset should fix `N2`, not sure how to process 1000s of villages like this

**Comment by cKellyDesign** [comment]
>Will we have a giant config file that has every single layer mapped to an OpenSRP jurisdiction id?

@moshthepitt - at the very least we need a config listing out source information for every tileset being used.

**Comment by moshthepitt** [comment]
Really curious to see what kind of configs will be needed for this, to prevent clunkiness/ambiguity.

**Comment by cKellyDesign** [comment]
@moshthepitt this is was my initial stab at configuration: https://github.com/onaio/reveal-frontend/blob/master/src/configs/settings.ts#L773-L806

```
/** interface descbribing basic country level information */
export interface JurisdictionsByCountry extends ADMN0 {

  // the GPS extents of given geometry(s)
  bounds?: any[];

  // the top level jurisdiction_Ids from OpenSRP
  // this is most useful for instances where tilesets DO NOT match the OpenSRP hierarchy
  jurisdictionIds: string[];

  // the top level jurisdiction_id of the country
  jurisdictionId: string;

  // the UUID of the Jurisdiction from the
  id?: string;

  // list of tilesets used for administrative boundaries
  tilesets?: Array<{
    idField: string; // the feature property cooresponding with jurisdiction_id (for joining)
    layer: string; // the Mapbox tileset-layer name
    parentIdField: string; // the feature property cooresponding with parent_id (for joining)
    url: string; // the Mapbox tileset url
  }>;
}
```

This specification emerged from the process of developing the IRS Jurisdiction selection page under shifting contexts with various available resources and still doesn't cover everything it needs to. Also the naming is terrible, sorry about that.

In my opinion, this needs some proper time, attention, and collaboration to get it to a place where it makes more sense (though I don't know when that will be possible).

**Comment by cKellyDesign** [comment]
The webapp needs to support a variety of contexts, but should have an optimal configuration and protocol, all contingencies should use a universal configuration alongside page-level business logic to generate layers for Gisida to render.

**Optimal Configuration**
* O1. Client provided Tilesets are used for all geometries
* O2. Location hierarchy is derived independent of the Map via Superset/OpenSRP API
* O3. GoeJSON is only used for structures

**Fallback Contingencies** 
* F1. Good case: Tileset are only abavailable for Operational Areas
* F2. Bad (current) case: Tilesets for Operational Areas are not available
* F3. Worst case: No tilesets are available (usually for non-country level 0 locations)



**F1 Requirements**
1. Client is ok with using HDX tilesets for administrative boundaries
2. Operational Area Parent Locations in OpenSRP have a data prop which matches the lowest level Administrative tileset `sprayAreasById[sprayArea.parentId].name === admin2byId[feature].name`
3. Webapp uses tilesets for all Gisida layers

**F2 Requirements**
1. Same as `F1.1`
2. Same as `F1.2`
3. WebApp progressively pulls all Operational Area GeoJSON from OpenSRP API to use in Gisida layers

**F3 Requirements**
1. WebApp progressively pulls Administrative GeoJson from OpenSRP API to use in Gisida layers
2. WebApp progressively pulls Operational Area GeoJSON from OpenSRP API to use in Gisida layers

**Comment by moshthepitt** [comment]
Fixed by #379
